### PR TITLE
Merge JUnit results from extended suites

### PR DESCRIPTION
**Pull request by smarterclayton**
Ensures that Jenkins does not double count skips until https://github.com/jenkinsci/junit-plugin/pull/54 is merged

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
[merge] only flakes now

File: tools/junitmerge/junitmerge.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+			// always replace a skip with a non-skip
+			r.runs[testCase.Name] = testCase
+		case existing.FailureOutput == nil && testCase.FailureOutput != nil:
+			// replace a passing test with a failing test
+			r.runs[testCase.Name] = testCase
+		}
+	}
+}
+
+func main() {
```
Comment (stevekuznetsov) [inline tools/junitmerge/junitmerge.go]: Why not `junitreport merge` a la `junitreport summarize`?

File: test/extended/core.sh [hunk 2]
```diff
@@ -34,4 +34,6 @@ os::log::info ""
 os::log::info "Running serial tests"
 FOCUS="${sf}" SKIP="${ss}" TEST_REPORT_FILE_NAME=core_serial os::test::extended::run -- -ginkgo.noColor -ginkgo.v -test.timeout 2h ${TEST_EXTENDED_ARGS-} || exitstatus=$?
 
+os::test::extended::merge_junit
```
Comment (stevekuznetsov) [inline test/extended/core.sh]: Should this be trapped so it runs on failure?

File: test/extended/setup.sh [hunk 3]
```diff
@@ -27,6 +27,7 @@ function os::test::extended::setup () {
 	os::util::ensure::built_binary_exists 'openshift'
 	os::util::ensure::built_binary_exists 'oadm'
 	os::util::ensure::built_binary_exists 'oc'
+	os::util::ensure::built_binary_exists 'junitmerge' 'tools/junitmerge'
```
Comment (stevekuznetsov) [inline test/extended/setup.sh]: Shouldn't need to specify the path for this, it looks into `tools/`

File: test/extended/core.sh (hunk 2 above)
Comment (smarterclayton) [inline test/extended/core.sh]: we already trap exit code

File: test/extended/setup.sh (hunk 3 above)
Comment (smarterclayton) [inline test/extended/setup.sh]: will do in a follow up

**Pull request merged**